### PR TITLE
[continuous-integration] move android check to GitHub Actions

### PR DESCRIPTION
**Pull request by bukepo**

### CODE REVIEW

**Comment by bukepo** [comment]
From https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on, 14.04 is not supported. I'll have to create a docker for this check.

**Comment by bukepo** [comment]
The docker image is about 755MB after some cleaning ups. I can continue to reduce the size. And I am going to publish the doctor image to the Docker Hub under openthread. What do you think? @jwhui @abtink

**Comment by jwhui** [comment]
> The docker image is about 755MB after some cleaning ups. I can continue to reduce the size. And I am going to publish the doctor image to the Docker Hub under openthread. What do you think? @jwhui @abtink

Sounds good to me!

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -265,3 +265,11 @@ jobs:
...
       run: |
         script/check-simulation-build
+
+  android:
+    runs-on: ubuntu-18.04
+    steps:
+    - uses: actions/checkout@v2
+    - name: Build
+      run: |
+        docker run --rm -it -v $PWD:/build/openthread openthread/android-trusty /build/openthread/script/check-android-build
```
Comment (bukepo) [inline .github/workflows/build.yml:275]:
```suggestion
        docker run --rm -v $PWD:/build/openthread openthread/android-trusty /build/openthread/script/check-android-build
```

File: tests/toranj/README.md [hunk 2]
```diff
@@ -7,14 +7,14 @@
...
-`toranj` tests will run as part of travis pull request validation in OpenThread and/or `wpantund` GitHub projects.
+`toranj` tests will run as part of GitHub Actions pull request validation in OpenThread and/or `wpantund` GitHub projects.
 
 
 ## Setup
 
 `toranj` requires `wpantund` to be installed.
 - Please follow [`wpantund` installation guide](https://github.com/openthread/wpantund/blob/master/INSTALL.md#wpantund-installation-guide). Note that `toranj` expects `wpantund` installed from latest master branch.
-- Alternative way to install `wpantund` is to use the same commands from travis `before_install` [script](https://github.com/openthread/openthread/blob/ef940b06594b370176b724733dbd7ad09617a693/.travis/before_install.sh#L110-L127) for build target `toranj-test-framework`.
+- Alternative way to install `wpantund` is to use the same commands from git workflow [Simulation](https://github.com/openthread/openthread/blob/master/.github/workflows/simulation.yml#L336-L341) for build target `toranj-test-framework`.
```
Comment (jwhui) [inline tests/toranj/README.md:17]:
```suggestion
- Alternative way to install `wpantund` is to use the same commands from git workflow [Simulation](https://github.com/openthread/openthread/blob/4b55284bd20f99a88e8e2c617ba358a0a5547f5d/.github/workflows/simulation.yml#L336-L341) for build target `toranj-test-framework`.
```

**Pull request merged**